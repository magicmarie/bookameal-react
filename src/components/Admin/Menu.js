import React from "react";
import { Link } from "react-router-dom";
import Delete from "./DeleteFromMenu";

// menu meal card: admin side
const Menu = props => (
  <div className="card mb-2 meal">
    <div className="card-body text-center">
      <div className="card-text"> {props.mealName}</div>
      <div className="card-text">{props.price}</div>
      <hr />
      <button
        to="#"
        className="fa fa-trash icon mealcard"
        data-toggle="modal"
        title="delete meal"
        data-target={`#deleteFromMenuModal${props.id}`}
      />
      <Delete
        id={props.id}
        mealName={props.mealName}
        ConfirmDeleteMeal={props.ConfirmDeleteMeal}
      />
    </div>
  </div>
);
export default Menu;
