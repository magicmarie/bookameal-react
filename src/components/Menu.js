import React from "react";
import { Link } from "react-router-dom";

//menu meal card: admin side
const Menu = props => (
  <div className="card mb-2 meal" style={{ width: "12rem", height: "8rem" }}>
    <div className="card-body text-center">
      <div className="card-text"> {props.mealName}</div>
      <div className="card-text">{props.price}</div>
      <div className="row">
        <Link
          to="#"
          className="btn btn-primary btn-sm centered"
          data-toggle="modal"
          data-target="#orderModal"
          onClick={() =>
            props.handleAddOrder(props.id, props.mealName, props.price)
          }
        >
          Add
        </Link>
      </div>
    </div>
  </div>
);
export default Menu;
