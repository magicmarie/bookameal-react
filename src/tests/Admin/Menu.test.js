import React from "react";
import { shallow } from "enzyme";
import Menu from "../../components/Admin/Menu";

describe("AdminMenuCard component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Menu />);
  });
  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
