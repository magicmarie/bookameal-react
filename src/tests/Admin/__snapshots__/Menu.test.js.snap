// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminMenuCard component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Menu />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card-body text-center"
>
        <div
                className="card-text"
        >
                 
        </div>
        <div
                className="card-text"
        />
        <hr />
        <button
                className="fa fa-trash icon mealcard"
                data-target="#deleteFromMenuModalundefined"
                data-toggle="modal"
                title="delete meal"
                to="#"
        />
        <DeleteMealFromMenu
                ConfirmDeleteMeal={undefined}
                id={undefined}
                mealName={undefined}
        />
</div>,
      "className": "card mb-2 meal",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card-text"
>
             
</div>,
          <div
            className="card-text"
/>,
          <hr />,
          <button
            className="fa fa-trash icon mealcard"
            data-target="#deleteFromMenuModalundefined"
            data-toggle="modal"
            title="delete meal"
            to="#"
/>,
          <DeleteMealFromMenu
            ConfirmDeleteMeal={undefined}
            id={undefined}
            mealName={undefined}
/>,
        ],
        "className": "card-body text-center",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              undefined,
            ],
            "className": "card-text",
          },
          "ref": null,
          "rendered": Array [
            " ",
            undefined,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "card-text",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fa fa-trash icon mealcard",
            "data-target": "#deleteFromMenuModalundefined",
            "data-toggle": "modal",
            "title": "delete meal",
            "to": "#",
          },
          "ref": null,
          "rendered": null,
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "ConfirmDeleteMeal": undefined,
            "id": undefined,
            "mealName": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card-body text-center"
>
          <div
                    className="card-text"
          >
                     
          </div>
          <div
                    className="card-text"
          />
          <hr />
          <button
                    className="fa fa-trash icon mealcard"
                    data-target="#deleteFromMenuModalundefined"
                    data-toggle="modal"
                    title="delete meal"
                    to="#"
          />
          <DeleteMealFromMenu
                    ConfirmDeleteMeal={undefined}
                    id={undefined}
                    mealName={undefined}
          />
</div>,
        "className": "card mb-2 meal",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="card-text"
>
               
</div>,
            <div
              className="card-text"
/>,
            <hr />,
            <button
              className="fa fa-trash icon mealcard"
              data-target="#deleteFromMenuModalundefined"
              data-toggle="modal"
              title="delete meal"
              to="#"
/>,
            <DeleteMealFromMenu
              ConfirmDeleteMeal={undefined}
              id={undefined}
              mealName={undefined}
/>,
          ],
          "className": "card-body text-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                undefined,
              ],
              "className": "card-text",
            },
            "ref": null,
            "rendered": Array [
              " ",
              undefined,
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "card-text",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fa fa-trash icon mealcard",
              "data-target": "#deleteFromMenuModalundefined",
              "data-toggle": "modal",
              "title": "delete meal",
              "to": "#",
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "ConfirmDeleteMeal": undefined,
              "id": undefined,
              "mealName": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
